Backend/editor/data/traversal.go

```diff
@@ -0,0 +1,84 @@
+package data
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"cms.csesoc.unsw.edu.au/editor/data/datamodels"
+)
+
+// Traverse returns the second last value and the last value pointed at by a specific path
+func Traverse(document datamodels.DataModel, subpaths []string) (reflect.Value, reflect.Value, error) {
+	prev := reflect.Value{}
+	curr := reflect.ValueOf(document)
+
+	for len(subpaths) > 0 {
+		prev, curr, subpaths = consumeField(prev, curr, subpaths)
+		prev, curr, subpaths = tryConsumeArrayElement(prev, curr, subpaths)
+
+		if !prev.IsValid() || !curr.IsValid() {
+			return reflect.Value{}, reflect.Value{}, fmt.Errorf("invalid path, couldn't find subpath %s", subpaths[0])
```

Comment:
A couple error messages were removed in this refactor so would be useful to have more information here for debugging / logging, esp if there's multiple subpaths with the same name. 

Maybe make a copy of subpaths like `remainingSubpath` and which is consumed by the above "consume" functions, and then you can do something like 

```go
for len(remainingSubpath) > 0 {
    // consume
    if someError {
        currIndex := len(subpaths) - len(remainingSubpaths) // maybe this is 1 off idk
        err := fmt.Error("invalid path, couldn't find subpath %s", strings.Join(subpaths[:currIndex], "/");
        return ..., err
}